compiler/plugins/target/AMD-AIE/iree-amd-aie/IR/AMDAIETypes.td

```diff
@@ -15,6 +15,12 @@ class AMDAIEDialect_Type<string name, string typeMnemonic, list<Trait> traits =
   let mnemonic = typeMnemonic;
 }
 
+def AMDAIE_AsyncTokenType : AMDAIEDialect_Type<"AsyncToken", "async_token">;
+def AMDAIE_AsyncSourceTokenType : AMDAIEDialect_Type<"AsyncSourceToken", "async_source_token">;
+def AMDAIE_AsyncTargetTokenType : AMDAIEDialect_Type<"AsyncTargetToken", "async_target_token">;
```

Comment:
Is there no way to add a brief description about them in the `.td` file? I understand `async_token` has correspondence with GPU dialect, so I had to read about it from the corresponding `.td` file to understand - hence asking.

And from what I understood :-
Presence of only `async_source_token` means the dma transfer execution from source side has begun and shouldn't be blocking.
Presence of only `async_target_token` means the dma transfer execution from target side has begun and shouldn't be blocking.

Presence of `async_source_token` and `async_target_token` means the same but now the dma transfer is bi-directional in nature ? I am trying to understand with an example.

Reply:
>Presence of only async_source_token means the dma transfer execution from source side has begun and shouldn't be blocking.

No, it means that the DMA transfer operation isn't be blocking, but a token is returned when the source DMA is done executing. Similarly, `async_target_token `, means that the DMA operation isn't blocking, but a token is returned when the target DMA is done executing. Having both `async_source_token` and `async_target_token` means that both the source and target DMAs will be returning a token that can be waited for.

Reply:
> Is there no way to add a brief description about them in the .td file?

I added more information to the descriptions and comments in the `AMDAIEOps.td` and `AMDAIETypes.td` files.